nearpy/tests/engine_tests.py

```diff
@@ -73,5 +73,58 @@ def test_retrieval_sparse(self):
             self.assertEqual(y_data, x_data)
             self.assertAlmostEqual(y_distance, 0.0, delta=delta)
 
+    def test_delete_vector_single_hash(self):
+        dim = 5
+        name_hash = 'testHash'
+        unibucket = UniBucket(name_hash)
+        engine = Engine(dim, lshashes=[unibucket])
+        # Index 20 random vectors (set their data to a unique string)
+        for index in range(20):
+            v = numpy.random.randn(dim)
+            engine.store_vector(v, index)
+        # Create random query vector
+        keys = set()
+        bucket = engine.storage.buckets[name_hash][name_hash]
+        keys = {i[1] for i in bucket}
+
+        engine.delete_vector(15) #delete the vector with key = 15
+
+        new_keys = set()
+        bucket = engine.storage.buckets[name_hash][name_hash]
+        new_keys = {i[1] for i in bucket}
+
+        self.assertTrue(len(keys) > len(new_keys))  #new keys has 19 elements instead of 20
+        self.assertIn(15, keys)
+        self.assertNotIn(15, new_keys) # the key 15 is the one missing
+
+    def test_delete_vector_multiple_hash(self):
+        dim = 5
+        hashes = []
+        for k in range(10): # 10 hashes
+            unibucket = UniBucket('name_hash_%d' % k)
+            hashes.append(unibucket)
+        engine = Engine(dim, lshashes=hashes)
+        # Index 20 random vectors (set their data to a unique string)
+        for index in range(20):
+            v = numpy.random.randn(dim)
+            engine.store_vector(v, index)
+        # Create random query vector
+        keys = set()
+        for lshash in engine.lshashes:
+            bucket = engine.storage.buckets[lshash.hash_name][lshash.hash_name]
+            keys = {i[1] for i in bucket}
```

Comment:
You rewrite `keys` here. I think it shoud be `keys.update(keys_from_bucket)` instead.